runtime/rasdump/javadump.cpp

```diff
@@ -5760,41 +5778,68 @@ regionIteratorCallback(J9JavaVM* virtualMachine, J9MM_IterateRegionDescriptor* r
 }
 
 #if JAVA_SPEC_VERSION >= 21
+static jvmtiIterationControl
+vthreadCountIteratorCallback(J9VMThread *vmThread, J9MM_IterateObjectDescriptor *object, void *userData) {
+	U_32 *vthreadCounts = (U_32 *)userData;
+	j9object_t vthread = J9VMJDKINTERNALVMCONTINUATION_VTHREAD(vmThread, object->object);
+	UDATA vThreadState = J9VMJAVALANGVIRTUALTHREAD_STATE(vmThread, vthread);
+#ifndef JVMTI_VTHREAD_STATE_STARTED
+#define JVMTI_VTHREAD_STATE_STARTED 1
+#endif
+	if (vThreadState <= JVMTI_VTHREAD_STATE_STARTED) {
```

Comment:
This should be
```c
	if (JVMTI_VTHREAD_STATE_NEW == vThreadState) {
```
to match the interpretation on line 2179.

Reply:
My understanding is that javacore files are used for debugging purpose which should reflect the actual JVM runtime status rather than the Java state. From a JVM level, new/started vthread both have identical native representation (ie. no J9VMContinuation structure associated, and no stack attached).

The `JVMTI_VTHREAD_STATE_STARTED` state is somewhat confusing as it really meant vthreads that have been scheduled to start but have not actually started.

I've added a comment section above this to explain the difference between Java/VM view on vthread state.

Reply:
I saw that comment: that doesn't address the conflict between the names used on lines 2179 and 2181 and the condition here.

Reply:
Have you considered distinguishing all virtual thread states?

Reply:
>I saw that comment: that doesn't address the conflict between the names used on lines 2179 and 2181 and the condition here.

I will update the use of name to "running thread" to be more reflective of the state name.
>Have you considered distinguishing all virtual thread states?

I don't think there is any real use to distinguish all different state in the overview as from VM's perspective, the key part of virtual thread is the `J9VMContinuation` structure which holds all the critical data. This can only be in 2 states, `NULL` if the vthread have not yet been mounted for the first time; a valid pointer to a `J9VMContinuation` after the vthread have been first mounted by VM.

I can include the vthread's state info as part of the individual information if that is something you feel important.